File: generator/src/main/java/ch/bailu/gtk/tag/EnumerationTag.java
package ch.bailu.gtk.tag;

import java.io.IOException;
import java.util.List;

public class EnumerationTag extends TypeTag {

    private final TagList<ParameterTag> members = new TagList<>();

    public EnumerationTag(Tag parent) {
        super(parent);
    }

    @Override
    public Tag getChild(String name, String prefix) {
        if ("member".equals(name)) {
            return members.addTag(new MemberTag(this));
        }
        return super.getChild(name, prefix);
    }

    @Override
    public void end() throws IOException {
        getBuilder().buildEnumeration(this);
    }

    public List<ParameterTag> getMembers() {
        return members;
    }
}

File: generator/src/main/kotlin/ch/bailu/gtk/parser/tag/EnumerationTag.kt
package ch.bailu.gtk.parser.tag

import java.io.IOException

class EnumerationTag(parent: TagWithParent) : TypeTag(parent) {
    private val members = TagList<ParameterTag>()

    override fun getChild(name: String, prefix: String): TagWithParent {
        return if ("member" == name) {
            members.addTag(MemberTag(this))
        } else super.getChild(name, prefix)
    }

    @Throws(IOException::class)
    override fun end() {
        getBuilder().buildEnumeration(this)
    }

    fun getMembers(): List<ParameterTag> {
        return members
    }
}
